pkg/features/kube_features.go

```diff
@@ -344,6 +344,12 @@ const (
 	//
 	// Enables CSI to use raw block storage volumes
 	CSIBlockVolume utilfeature.Feature = "CSIBlockVolume"
+
+	// owner: @xing-yang
+	// alpha: v1.12
+	//
+	// Enable volume snapshot data source support.
+	VolumeSnapshotDataSource utilfeature.Feature = "VolumeSnapshotDataSource"
```

Comment:
We need to add validation for data source. The kind, API group etc.

Reply:
Sure

Reply:
Somewhere you need to test the field gate and nil the field if it is not enabled.  See other alpha fields for examples.

Reply:
Done